apis/field_error.go

```diff
@@ -139,10 +153,54 @@ func (fe *FieldError) getNormalizedErrors() []FieldError {
 	for _, e := range fe.errors {
 		errors = append(errors, e.getNormalizedErrors()...)
 	}
-	sort.Slice(errors, func(i, j int) bool { return errors[i].Message < errors[j].Message })
+
 	return errors
 }
 
+// merge will combine errors that differ only in their paths.
+// This assumes that errors has been normalized.
+func merge(errors []FieldError) []FieldError {
+	if len(errors) <= 1 {
+		return errors
+	}
+
+	// Sort first.
+	sort.Slice(errors, func(i, j int) bool { return errors[i].Message < errors[j].Message })
```

Comment:
I suspect this would be more clearly implemented as a `map[string /* Message */]FieldError` to build up the deduplicated `FieldErrors` and then sorting the result.

Would it be useful to *store* things as a `map` or "set" internally?

Reply:
Let me try.